src/request.ts

```diff
@@ -152,6 +169,108 @@ export async function getRequestLinkDetails({
 	return x
 }
 
+export async function prepareXchainRequestFulfillmentTransaction({
+	senderAddress,
+	recipientAddress,
+	destinationChainId,
+	destinationToken,
+	fromToken,
+	fromTokenDecimals,
+	fromAmount,
+	fromChainId,
+	link,
+	squidRouterUrl,
+	provider,
```

Comment:
why not calculate fromAmount within this function? Why is fromAmount a parameter?